image2tensor/src/image2tensor.rs

```diff
@@ -1,82 +1,97 @@
-use image::DynamicImage;
-use image::io::Reader;
+use image::{io::Reader, Bgr, DynamicImage, ImageBuffer, Rgb};
 
 #[derive(Debug, Copy, Clone)]
 pub enum TensorType {
     F16,
     F32,
     U8,
-    I32
+    I32,
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum ColorOrder {
+    RGB,
+    BGR,
 }
 
 // Take the image located at 'path', open it, resize it to height x width, and then converts
 // the pixel precision to the type requested.
-pub fn convert_image (
-    path: String,
+pub fn convert_image(
+    path: &String,
     width: u32,
     height: u32,
     precision: TensorType,
-    out_buff: &mut [u8]
-    ) -> i32 {
+    order: ColorOrder,
+) -> Vec<u8> {
     let raw_file = Reader::open(path);
     let raw_file = match raw_file {
         Ok(file) => file,
         Err(e) => {
             println!("Failed to open the file: {:?}", e);
-            return -1;
-        },
+            return vec![];
+        }
     };
 
-    let decoded = raw_file.decode(); //.unwrap();
+    let decoded = raw_file.decode();
     let decoded = match decoded {
         Ok(pixels) => pixels,
         Err(e) => {
             println!("Failed to decode the image: {:?}", e);
-            return -1;
-        },
+            return vec![];
+        }
     };
 
     let dyn_img: DynamicImage = decoded.resize_exact(width, height, image::imageops::Triangle);
-    // Switch from RGB to BGR
-    let bgr_img = dyn_img.to_bgr8();
+    let rgb_bind: ImageBuffer<Rgb<u8>, Vec<u8>>;
+    let bgr_bind: ImageBuffer<Bgr<u8>, Vec<u8>>;
```

Comment:
No need to declare these up here. They can go inside the `match` arms and I suspect the explicit type may not be needed.

Reply:
I'll simplify it to this, then I don't need either bindings
    match order {
        ColorOrder::RGB => {
            save_bytes(dyn_img.to_rgb8().as_raw(), precision)
        }
        ColorOrder::BGR => {
            save_bytes(dyn_img.to_bgr8().as_raw(), precision)
        }
    }